ksql-engine/src/main/java/io/confluent/ksql/util/ExecutorWithRetries.java

```diff
@@ -0,0 +1,94 @@
+/*
+ * Copyright 2017 Confluent Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ **/
+
+package io.confluent.ksql.util;
+
+import org.apache.kafka.common.errors.RetriableException;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import java.util.concurrent.Callable;
+import java.util.concurrent.CompletableFuture;
+import java.util.concurrent.Future;
+import java.util.function.Supplier;
+
+public final class ExecutorWithRetries {
```

Comment:
Can you rename to something ending in Util (e.g. ExecuteWithRetriesUtil) since this is a purely static class?

Reply:
Maybe rename to `ExecutorUtil` and rename the methods `executeWithRetries`. That way, the code still reads well if using static imports.